### DISCUSSION TITLE
Histogram_quantile

### DISCUSSION THREAD

**Discussion by mkori** [prometheus/prometheus#10897] [General]
histogram_quantile(0.01, sum by (job, le) (rate(apiserver_client_certificate_expiration_seconds_bucket{job="apiserver"}[5m])
The above query always show value 31104000 in seconds which is 360 days.
Why is the value not changing if the seconds to apiserver certificate expire is increasing.
What does this histogram_quantile 0.01 calculates ?

**Comment by beorn7** [comment]
> What does this histogram_quantile 0.01 calculates ?

It shows you the 1st percentile of expiration time. Or in other words: If you only look at the 1% certificates that have the fewest expiration seconds left, it will tell you the largest value of those 1%.

I don't know what the bucket layout in that histogram is and how your certificates are distributed. My wild guess is that the largest bucket is 31104000 seconds AKA 360 days, and that _all_ your certificates have more than 31104000 seconds left until they expire. Then even the 1% soonest-to-expire certificates will have 31104000 seconds left. And since 31104000 seconds is (presumably) the largest value the histogram can track, that's the value you get in the result.
